src/reducers/index.ts

```diff
@@ -54,6 +54,13 @@ function reduce(state: StoreState, action: Action) {
     let { matchesList, ...rest } = state;
     return { matchesList: action.setMatchesList, ...rest };
 
+  } else if (action.setSignals) {
+    let { signals, ...rest } = state;
+    return { signals: action.setSignals, ...rest };
+
+  } else if (action.setMyUser) {
+    let { myUser, ...rest } = state;
+    return { myUser: action.setMyUser, ...rest };
```

Comment:
Direct syntax again:

```
return {...state, myUser: action.setMyUser }
```

You don't need any other line.

Reply:
Nice trick, correct :)

Reply:
Hi I have one question here. 
For this expected state, if we try to change the value of my user in the default state, shouldn't we keep the same key? 
So for here, do we need to write 
let expectedState = Object.assign(storeStateDefault, {myUser: userDetails}); ?
Because "myUser" is the key in the default state.

Reply:
You take the previous state's all key by using `...state` and you replace the keys you want to change by specifying them as such in the object literal constructor pattern.

So `...state` will introduce all of the other keys in the new object (whenever you use `{}` it creates a new object) and the keys which need to be replaced are specified one by one.